drivers/clock_control/clock_control_nrf2_common.h

```diff
@@ -79,10 +79,4 @@ struct clock_lrcconf_sink {
 	sys_snode_t node;
 };
 
-/**
- * @brief Request or release lrcconf main power domain
- */
-void clock_request_lrcconf_poweron_main(struct clock_lrcconf_sink *sink);
```

Comment:
Please remove also the `struct clock_lrcconf_sink` definition. There is no need to keep it as it can be replaced with just `sys_snode_t`.